07-reproj.qmd

```diff
@@ -86,7 +87,74 @@ Let's look at how CRSs are stored in Python spatial objects and how they can be
 world.crs
 ```
 
-...
+The output specifies the following pieces of information:
+
+1. The CRS type (`Geographic 2D CRS`) and EPSG code (`EPSG:4326`) 
+1. The CRS name (`WGS 84`)
+1. The axes (`latitude`, `longitude`) and their units (`degree`)
+1. The applicable area name (`World`) and bounding box (`(-180.0, -90.0, 180.0, 90.0)`)
+1. The datum (`WGS 84`)
+
+The WKT representation, which is internally used when saving the object to a file or doing any coordinate operations, can be extracted using `.crs.to_wkt()` as shown above (@sec-coordinate-reference-systems). Above, we can see that the `world` object has the WGS84 ellipsoid, uses the Greenwich prime meridian, and the latitude and longitude axis order. We also have the suitable suitable area specification for the use of this CRS, and CRS identifier: `EPSG:4326`.
+
+The CRS specification object, such as `world.crs`, has several other useful properties and methods to retrieve additional information about the used CRS. For example, try to run:
+
+* `world.crs.is_geographic` to check is the CRS is geographic or not
+* `world.crs.axis_info[0].unit_name` and `world.crs.axis_info[1].unit_name` to find out the CRS units of both axes (typically having identical units)
+* `world.crs.to_authority()` extracts the authority (e.g., `EPSG`) and the identifier (e.g., `4326`)
+* `world.crs.to_proj4()` returns the proj-string representation
+
+In cases when a coordinate reference system (CRS) is missing or the wrong CRS is set, the `.set_crs` method can be used on a `GeoSeries` or a `GeoDataFrame` to set it. The CRS can be specified using an EPSG code as the first argument. In case the object already has a different CRS definition, we must also specify `allow_override=True` to replace it (otherwise we get an error). For example, here we set the `EPSG:4326` CRS, which has no effect because `world` already has that exact CRS definition:
+
+```{python}
+world2 = world.set_crs(4326)
+world2
+```
+
+and here we replace the definition from the existing `EPSG:4326`, to a new definition `EPSG:3857`:
```

Comment:
@michaeldorman and what do to if we had data with CRS of some other "authority", e.g., ESRI:54009?

Reply:
> @michaeldorman and what do to if we had data with CRS of some other "authority", e.g., ESRI:54009?

Good point, thanks! Now added an example: 
`world.set_crs('ESRI:54009', allow_override=True)`